### PR TITLE
Use curl in favor of wget

### PR DESCRIPTION
**Pull request by thisisjoshford**
As wget does not ship with macOS we should use a tool that is more common across multiple operating systems. [curl ](https://curl.se/) comes with most Linux distros, ships with macOS, as well as Windows 10.

Fixes https://github.com/near/near-sdk-js/issues/160

### CODE REVIEW

**Comment by volovyk-s** [comment]
@thisisjoshford looks like your branch is outdated,
The current script with wget is in a JS file now. JSVM tests and logic are not used at the moment.

**Comment by thisisjoshford** [comment]
> @thisisjoshford looks like your branch is outdated, The current script with wget is in a JS file now. JSVM tests and logic are not used at the moment.

Was just trying to figure out why I'm off and noticed that I'm going off of `master` and we switched to `develop` as default, correct @volovyk-s ?

**Comment by gagdiez** [comment]
@thisisjoshford I think you are missing the `-L -o` flags in the commands

File: cli/post-install.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ const BINARYEN_SYSTEM_NAME = OS === 'Linux' ? 'Linux' : OS === 'Darwin' ? 'macOS
 const BINARYEN_ARCH_NAME = ARCH === 'x86_64' ? 'X64' : ARCH === 'arm64' ? 'arm64' : 'other';
 const BINARYEN_TAR_NAME = `binaryen-${BINARYEN_SYSTEM_NAME}-${BINARYEN_ARCH_NAME}.tar.gz`;
 
-await executeCommand(`wget https://github.com/near/binaryen/releases/download/${BINARYEN_VERSION_TAG}/${BINARYEN_TAR_NAME}`);
+await executeCommand(`curl https://github.com/near/binaryen/releases/download/${BINARYEN_VERSION_TAG}/${BINARYEN_TAR_NAME}`);
```
Comment (gagdiez) [inline cli/post-install.js:23]: @thisisjoshford 
curl -L -o  ${BINARYEN_TAR_NAME} https://github.com/near/binaryen/releases/download/${BINARYEN_VERSION_TAG}/${BINARYEN_TAR_NAME}

Comment (gagdiez) [reply cli/post-install.js:23]: giving that you will need to apply this to all files, maybe you could make an aux method `download`, which creates the command for you.

Comment (thisisjoshford) [reply cli/post-install.js]: haha was just troubleshooting this issue and pushed a fix then saw this comment.

**Comment by gagdiez** [comment]
@thisisjoshford @volovyk-s  should we further add `-s` to keep it silent?

**Comment by thisisjoshford** [comment]
> @thisisjoshford @volovyk-s should we further add `-s` to keep it silent?

Sounds good, let me add it.

**Review by ailisp — APPROVED** [review]
Looks good to me!

**Comment by maxhr** [comment]
@thisisjoshford please don't make it silent, as it will also silent errors, and the installation will fail on following lines without a clear reason.

File: cli/post-install.js [hunk 2]
```diff
@@ -20,7 +20,7 @@ const BINARYEN_SYSTEM_NAME = OS === 'Linux' ? 'Linux' : OS === 'Darwin' ? 'macOS
 const BINARYEN_ARCH_NAME = ARCH === 'x86_64' ? 'X64' : ARCH === 'arm64' ? 'arm64' : 'other';
 const BINARYEN_TAR_NAME = `binaryen-${BINARYEN_SYSTEM_NAME}-${BINARYEN_ARCH_NAME}.tar.gz`;
 
-await executeCommand(`wget https://github.com/near/binaryen/releases/download/${BINARYEN_VERSION_TAG}/${BINARYEN_TAR_NAME}`);
+await executeCommand(`curl -LO https://github.com/near/binaryen/releases/download/${BINARYEN_VERSION_TAG}/${BINARYEN_TAR_NAME} -s`);
```
Comment (maxhr) [inline cli/post-install.js:23]: `-s` will also silent errors

**Review by maxhr — CHANGES_REQUESTED** [review]

**Comment by maxhr** [comment]
@thisisjoshford let's make it this way:
```
async function download(url) {
    await executeCommand(`curl -LOf ${url}`);
}
```
...
```
await download(`https://github.com/near/quickjs/releases/download/........``);
await download(......);
```

**Comment by maxhr** [comment]
@thisisjoshford since I don't have permissions to push to this repo, here is a PR to PR (which is ridiculous):
https://github.com/near/near-sdk-js/pull/167

**Review by gagdiez — APPROVED** [review]

File: cli/post-install.js (hunk 2 above)
Comment (thisisjoshford) [reply cli/post-install.js]: gotcha, nice catch!

**Pull request merged**